presto-session-property-managers/src/main/java/io/prestosql/plugin/session/db/DbSessionPropertyManager.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.session.db;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.plugin.session.SessionMatchSpec;
+import io.prestosql.spi.session.SessionConfigurationContext;
+import io.prestosql.spi.session.SessionPropertyConfigurationManager;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link SessionPropertyConfigurationManager} implementation that connects to a database for fetching information
+ * about session property overrides given {@link SessionConfigurationContext}.
+ */
+public class DbSessionPropertyManager
+        implements SessionPropertyConfigurationManager
+{
+    private final SessionPropertiesDao dao;
+    private final DbSpecsProvider specsProvider;
+
+    @Inject
+    public DbSessionPropertyManager(SessionPropertiesDao dao, DbSpecsProvider specsProvider)
+    {
+        this.dao = requireNonNull(dao, "SessionPropertiesDao is null");
+        this.specsProvider = requireNonNull(specsProvider, "specsProvider is null");
+    }
+
+    @PreDestroy
+    public void destroy()
+    {
+        specsProvider.destroy();
+    }
+
+    @PostConstruct
+    public void start()
```

Comment:
Nit: put the start method before the destroy method (that ordering makes more sense since they will be called in that order)